### DISCUSSION TITLE
Ideas for channel operator features

### DISCUSSION THREAD

**Discussion by ahmednaalik51** [BearWare/TeamTalk5#1455] [Ideas]
Hi,
I have some ideas for the operator feature in TeamTalk.
The first, if you are in for example, a server in the country on a public server list, and you can make temp channels, then I think that operator should be restored for the person who created the channel automaticly if they lost the connection and logged out of the server, and the channel is still available.
Yes, this can be done with server rights, and yes they can type in the op password but not everyone can make a server, and sometimes, it will be better to even just make a channel on a public server.
It can make sure the person is the same by checking the ip address, or if using pro server, then it can see the bearware account that created the channel and restore the operator back to that user if they come into the channel once more.
It is better, because if you are in a classroom and need to control your channel, then it will give you your rights back on it's own and you do not need to type the password each time, and it's also good for devices that don't support that feature.
This also good if you lost the connection, then you didn't need to worry about getting your operator rights removed, unless the channel was gone after the last person left, then you didn't become an operator anymore.
@bear101 What are you thinking about this idea? It would be a really good one if you can maybe implement into maybe the pro server.
If you want more informations, please let me know and I will provide.
Have a great day!

**Comment by bear101** [comment]
I like the idea to "resume" operator status using WebLogin. This is a way to ensure it's the correct user. Using an IP-address is not a secure way.

**Comment by bear101** [comment]
I've added the "resume operator" feature in this merge request: #1456

**Comment by alessio0292** [comment]
Hy Amed nice to meet you. I'd like speak with Bjoern bear101 that is very bad with me and I'd like a present org gyft from this person.

**Comment by bear101** [comment]
@ahmednaalik51 You can try out the new feature by downloading the TeamTalk Pro server from this [Ubuntu 22 Build](https://github.com/BearWare/TeamTalk5/actions/runs/2899314854).

**Reply by ahmednaalik51** [reply]
@bear101 This is awesome!
Will this be available in all the pro server versions or only the for Ubuntu?
Thanks for taking the time to implement my idea.

**Reply by bear101** [reply]
You can download for other platforms here: https://github.com/BearWare/TeamTalk5/actions?query=branch%3Amaster
